### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by skeletronys**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {"line": error["line_number"], "column": error["column_number"],
+            "message": error["text"], "name": error["code"],
+            "source": "flake8"}
```
Comment (Dimosphen1) [inline app/main.py]: Place each key on a new line to improve the readability, there should be no keys on the same line with `{` and `}`

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error)
+            for error in errors],
+            "path": file_path, "status": "passed"
+            if not errors else "failed"}
```
Comment (Dimosphen1) [inline app/main.py]: Place each key on a new line to improve the readability, there should be no keys on the same line with `{` and `}`

Comment (Dimosphen1) [inline app/main.py]: Rewrite `status` value if condition without using `not`

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [{"errors": [format_linter_error(error) for error in errors],
+             "path": file_path, "status": "passed"
+            if not errors else "failed"}
+            for file_path, errors in linter_report.items()]
```
Comment (Dimosphen1) [inline app/main.py]: Call `format_single_linter_file` inside of the `format_linter_report` to follow DRY principles.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

**Comment by skeletronys** [comment]
Fix all

**Pull request closed without merging**